File: src/main/java/com/heekng/celloct/dto/WorkUpdateRequestDto.java
package com.heekng.celloct.dto;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class WorkUpdateRequestDto {

    @Getter
    @Setter
    @NoArgsConstructor
    public static class AddRequest {
        private Long workId;
        private Long staffId;
        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
        private LocalDate workDate;
        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
        private LocalDateTime startDate;
        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
        private LocalDateTime endDate;
        private String note;

        @Builder
        public AddRequest(Long workId, Long staffId, LocalDate workDate, LocalDateTime startDate, LocalDateTime endDate, String note) {
            this.workId = workId;
            this.staffId = staffId;
            this.workDate = workDate;
            this.startDate = startDate;
            this.endDate = endDate;
            this.note = note;
        }
    }

    @Getter
    public static class updateRequest {
        private Long workUpdateRequestId;
        private LocalDate updateDate;
        private LocalDateTime updateStartDate;
        private LocalDateTime updateEndDate;

        @Builder
        public updateRequest(Long workUpdateRequestId, LocalDate updateDate, LocalDateTime updateStartDate, LocalDateTime updateEndDate) {
            this.workUpdateRequestId = workUpdateRequestId;
            this.updateDate = updateDate;
            this.updateStartDate = updateStartDate;
            this.updateEndDate = updateEndDate;
        }
    }


}

File: src/main/kotlin/com/heekng/celloct/dto/WorkUpdateRequestDto.kt
package com.heekng.celloct.dto

import com.heekng.celloct.entity.WorkUpdateRequest
import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate
import java.time.LocalDateTime

class WorkUpdateRequestDto {
    data class AddRequest (
        var workId: Long,
        var staffId: Long,
        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
        val workDate: LocalDate,
        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
        val startDate: LocalDateTime,
        @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
        val endDate: LocalDateTime,
        val note: String? = null
    )

    data class UpdateRequest (
        val workUpdateRequestId: Long,
        val updateDate: LocalDate,
        val updateStartDate: LocalDateTime,
        val updateEndDate: LocalDateTime
    )

    data class WorkUpdateRequestListResponse (
        val staffName: String,
        val workUpdateRequestId: Long,
        val workId: Long,
        val createDate: LocalDateTime,
        val workDate: LocalDate,
    ) {
        constructor(workUpdateRequest: WorkUpdateRequest): this(
            staffName = workUpdateRequest.work.staff.name,
            workUpdateRequestId = workUpdateRequest.id!!,
            workId = workUpdateRequest.work.id!!,
            createDate = workUpdateRequest.createDate!!,
            workDate = workUpdateRequest.work.workTime.workDate,
        )
    }
}